pkg/master/master.go

```diff
@@ -547,22 +553,36 @@ func (m *Master) removeThirdPartyStorage(path string) error {
 			return err
 		}
 		delete(m.thirdPartyResources, path)
-		m.RemoveAPIGroupForDiscovery(getThirdPartyGroupName(path))
 	}
 	return nil
 }
 
 // RemoveThirdPartyResource removes all resources matching `path`.  Also deletes any stored data
 func (m *Master) RemoveThirdPartyResource(path string) error {
+	group, kind := getThirdPartyGroupKind(path)
 	if err := m.removeThirdPartyStorage(path); err != nil {
 		return err
 	}
-
 	services := m.HandlerContainer.RegisteredWebServices()
 	for ix := range services {
-		root := services[ix].RootPath()
-		if root == path || strings.HasPrefix(root, path+"/") {
-			m.HandlerContainer.Remove(services[ix])
+		if strings.HasPrefix(services[ix].RootPath(), makeThirdPartyPath(group)) {
+			m.removeAllThirdPartyResourceRoutes(services[ix], kind)
+		}
+	}
+
+	len := 0
```

Comment:
Will be good to add a comment here explaining what this code is doing.
You can say something like "Determine if any resource exists in this group. We will delete the group if no resource exits".